poc/gen_test_vec.py

```diff
@@ -0,0 +1,395 @@
+        'public_share': public_share.hex(),
+        'keys': printable_keys,
+    }
+
+    os.system('mkdir -p {}'.format(TEST_VECTOR_PATH))
+    filename = '{}/{}_{}.json'.format(TEST_VECTOR_PATH, idpf.test_vec_name,
+                                      test_vec_instance)
+    with open(filename, 'w') as f:
+        json.dump(test_vec, f, indent=4, sort_keys=True)
+        f.write('\n')
+
+
+# XOF
+
+def gen_test_vec_for_xof(cls: type[Xof]) -> None:
+    seed = gen_rand(cls.SEED_SIZE)
+    dst = b'domain separation tag'
+    binder = b'binder string'
+    length = 40
+
+    test_vector = {
+        'seed': seed.hex(),
+        'dst': dst.hex(),
+        'binder': binder.hex(),
+        'length': length,
+        'derived_seed': None,  # set below
+        'expanded_vec_field128': None,  # set below
+    }
+
+    derived_seed = cls.derive_seed(seed, dst, binder).hex()
+    expanded_vec_field128 = Field128.encode_vec(
+        cls.expand_into_vec(Field128, seed, dst, binder, length)).hex()
+    test_vector['derived_seed'] = derived_seed
+    test_vector['expanded_vec_field128'] = expanded_vec_field128
+
+    print('{}:'.format(cls.test_vec_name))
+    print('  seed: "{}"'.format(test_vector['seed']))
+    print('  dst: "{}"'.format(test_vector['dst']))
+    print('  binder: "{}"'.format(test_vector['binder']))
+    print('  length: {}'.format(test_vector['length']))
+    print('  derived_seed: "{}"'.format(test_vector['derived_seed']))
+    print('  expanded_vec_field128: >-')
+    print_wrapped_line(expanded_vec_field128, tab=4)
+
+    os.system('mkdir -p {}'.format(TEST_VECTOR_PATH))
+    with open('{}/{}.json'.format(
+            TEST_VECTOR_PATH, cls.__name__), 'w') as f:
+        json.dump(test_vector, f, indent=4, sort_keys=True)
+        f.write('\n')
+
+
+if __name__ == '__main__':
+    from vdaf_poc import idpf_bbcggi21, vdaf_poplar1, vdaf_prio3, xof
+
+    # Prio3 variants
+    gen_test_vec_for_vdaf(vdaf_prio3.Prio3Count(2), None, [1], 0)
+    gen_test_vec_for_vdaf(vdaf_prio3.Prio3Count(3), None, [1], 1)
+    gen_test_vec_for_vdaf(vdaf_prio3.Prio3Sum(2, 8), None, [100], 0)
+    gen_test_vec_for_vdaf(vdaf_prio3.Prio3Sum(3, 8), None, [100], 1)
+    gen_test_vec_for_vdaf(
+        vdaf_prio3.Prio3SumVec(2, 10, 8, 9),
+        None,
+        [
+            list(range(10)),
+            [1] * 10,
+            [255] * 10
+        ],
+        0,
+    )
+    gen_test_vec_for_vdaf(
+        vdaf_prio3.Prio3SumVec(3, 3, 16, 7),
+        None,
+        [
+            [10000, 32000, 9],
+            [19342, 19615, 3061],
+            [15986, 24671, 23910]
+        ],
+        1,
+    )
+    gen_test_vec_for_vdaf(vdaf_prio3.Prio3Histogram(2, 4, 2),  None, [2], 0)
+    gen_test_vec_for_vdaf(vdaf_prio3.Prio3Histogram(3, 11, 3),  None, [2], 1)
+    gen_test_vec_for_vdaf(
+        vdaf_prio3.Prio3MultihotCountVec(2, 4, 2, 2),
+        None,
+        [[0, 1, 1, 0]],
+        0,
+    )
+
+    # Poplar1
+    tests = [
+        (0, [0, 1]),
```

Comment:
nit: the others use tuples, while this uses a list

Reply:
Done.